sql/sql_select.cc

```diff
@@ -8050,8 +8050,13 @@ best_access_path(JOIN      *join,
             }
 
             /* Limit the number of matched rows */
-            tmp= cost_for_index_read(thd, table, key, (ha_rows) records,
```

Comment:
`/server/sql/sql_select.cc:8054:48: runtime error: -nan is outside the range of representable values of type 'unsigned long long'`